tests/checkers/rst-yamllint.py

```diff
@@ -0,0 +1,317 @@
+
+        # Now that we have the offsets, we can actually do some processing...
+        if language not in {'YAML', 'yaml', 'yaml+jinja', 'YAML+Jinja'}:
+            if language is None:
+                self.__results.append({
+                    'path': self.__path,
+                    'line': row_offset + 1,
+                    'col': col_offset + 1,
+                    'message': f"Literal block without language!",
+                })
+                return
+            if language not in {'bash', 'ini', 'console', 'text', 'shell', 'shell-session', 'jinja', 'ansible-output', 'none', 'json', 'python', 'Jinja', 'diff', 'powershell', 'md', 'reStructuredText', 'rst', 'sh', 'Python', 'csharp'}:
+                self.__results.append({
+                    'path': self.__path,
+                    'line': row_offset + 1,
+                    'col': col_offset + 1,
+                    'message': f"Warning: literal block with disallowed language: {language}. If the language should be allowed, the checker needs to be updated.",
+                })
+            raise nodes.SkipNode
+
+        # So we have YAML. Let's lint it!
+        try:
+            conf = YamlLintConfig(YAMLLINT_CONFIG)
+            problems = linter.run(io.StringIO(node.rawsource.rstrip() + '\n'), conf, self.__path)
+            for problem in problems:
+                if problem.level not in REPORT_LEVELS:
+                    continue
+                msg = f"{problem.level}: {problem.desc}"
+                if problem.rule:
+                    msg += f"  ({problem.rule})"
+                self.__results.append({
+                    'path': self.__path,
+                    'line': row_offset + problem.line,
+                    'col': col_offset + problem.column,
+                    'message': msg,
+                })
+        except Exception as exc:
+            self.__results.append({
+                'path': self.__path,
+                'line': row_offset + 1,
+                'col': col_offset + 1,
+                'message': f"Internal error while linting YAML: exception {type(exc)}: {str(exc).replace("\n", " / ")}; traceback: {traceback.format_exc()!r}",
+            })
+
+        raise nodes.SkipNode
+
+
+def main():
+    paths = sys.argv[1:] or sys.stdin.read().splitlines()
+    results = []
+
+    for directive in (
+        'code',
+        'code-block',
+        'sourcecode',
+    ):
+        register_directive(directive, CodeBlockDirective)
+
+    # The following docutils directives should better be ignored:
+    for directive in (
+        'parsed-literal',
+    ):
+        register_directive(directive, IgnoreDirective)
+
+    # TODO: should we handle the 'literalinclude' directive? maybe check file directly if right extension?
+    # (https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude)
+
+    docs_root = os.path.normpath(os.path.join(os.path.dirname(__file__), '..', '..', 'docs', 'docsite', 'rst'))
```

Comment:
nit: try to default to pathlib objects everywhere

Reply:
3d6aaff4537283facda4348271966618395dacfc